libdevice/cmake/modules/SYCLLibdevice.cmake

```diff
@@ -22,266 +25,83 @@ set(compile_opts
+endfunction()
+
+function(add_devicelib_spv spv_filename)
+  cmake_parse_arguments(SPV  "" "" "SRC;DEP" ${ARGN})
+  set(devicelib-spv-file ${spv_binary_dir}/${spv_filename}.spv)
+  add_custom_command(OUTPUT ${devicelib-spv-file}
+                     COMMAND ${clang} -fsycl-device-only -fno-sycl-use-bitcode
+                             ${compile_opts}
+                             ${CMAKE_CURRENT_SOURCE_DIR}/${SPV_SRC}
+                             -o ${devicelib-spv-file}
+                     MAIN_DEPENDENCY ${SPV_SRC}
+                     DEPENDS ${SPV_DEP}
+                     VERBATIM)
+  set(devicelib-spv-target ${spv_filename}-spv)
+  add_custom_target(${devicelib-spv-target} DEPENDS ${devicelib-spv-file})
+  add_dependencies(libsycldevice-spv ${devicelib-spv-target})
+  install(FILES ${devicelib-spv-file}
+          DESTINATION ${install_dest_spv}
+          COMPONENT libsycldevice)
+endfunction()
+
+set(crt_obj_deps wrapper.h device.h spirv_vars.h sycl-compiler)
+add_devicelib_obj(libsycl-crt SRC crt_wrapper.cpp DEP ${crt_obj_deps})
+add_devicelib_obj(libsycl-fallback-cassert SRC fallback-cassert.cpp DEP ${crt_obj_deps})
+add_devicelib_obj(libsycl-fallback-cstring SRC fallback-cstring.cpp DEP ${crt_obj_deps})
+add_devicelib_spv(libsycl-fallback-cassert SRC fallback-cassert.cpp DEP ${crt_obj_deps})
+add_devicelib_spv(libsycl-fallback-cstring SRC fallback-cstring.cpp DEP ${crt_obj_deps})
+
+set(complex_obj_deps device_complex.h device.h sycl-compiler)
+add_devicelib_obj(libsycl-complex SRC complex_wrapper.cpp DEP ${complex_obj_deps})
+add_devicelib_obj(libsycl-complex-fp64 SRC complex_wrapper_fp64.cpp DEP ${complex_obj_deps} )
+add_devicelib_obj(libsycl-fallback-complex SRC fallback-complex.cpp DEP ${complex_obj_deps})
+add_devicelib_obj(libsycl-fallback-complex-fp64 SRC fallback-complex-fp64.cpp DEP ${complex_obj_deps} )
+add_devicelib_spv(libsycl-fallback-complex SRC fallback-complex.cpp DEP ${complex_obj_deps})
+add_devicelib_spv(libsycl-fallback-complex-fp64 SRC fallback-complex-fp64.cpp DEP ${complex_obj_deps})
+
+set(cmath_obj_deps device_math.h device.h sycl-compiler)
+add_devicelib_obj(libsycl-cmath SRC cmath_wrapper.cpp DEP ${cmath_obj_deps})
+add_devicelib_obj(libsycl-cmath-fp64 SRC cmath_wrapper_fp64.cpp DEP ${cmath_obj_deps} )
+add_devicelib_obj(libsycl-fallback-cmath SRC fallback-cmath.cpp DEP ${cmath_obj_deps})
+add_devicelib_obj(libsycl-fallback-cmath-fp64 SRC fallback-cmath-fp64.cpp DEP ${cmath_obj_deps})
+add_devicelib_spv(libsycl-fallback-cmath SRC fallback-cmath.cpp DEP ${cmath_obj_deps})
+add_devicelib_spv(libsycl-fallback-cmath-fp64 SRC fallback-cmath-fp64.cpp DEP ${cmath_obj_deps})
+
+set(itt_obj_deps device_itt.h spirv_vars.h device.h sycl-compiler)
+add_devicelib_obj(libsycl-itt-stubs SRC itt_stubs.cpp DEP ${itt_obj_deps})
+add_devicelib_obj(libsycl-itt-compiler-wrappers SRC itt_compiler_wrappers.cpp DEP ${itt_obj_deps})
+add_devicelib_obj(libsycl-itt-user-wrappers SRC itt_user_wrappers.cpp DEP ${itt_obj_deps})
```

Comment:
I think that having one extra function/macro/etc. that creates both spv and obj simultaneously will make this code easier to read. Can you please implement that?

Reply:
> I think that having one extra function/macro/etc. that creates both spv and obj simultaneously will make this code easier to read. Can you please implement that?

The function combining obj and spv creation is added and the grouping is updated too.

Thanks very much.